client/network/src/lib.rs

```diff
@@ -90,23 +90,49 @@
 //! ## Substreams
 //!
 //! Once a connection has been established and uses multiplexing, substreams can be opened. When
-//! a substream is open, the **multistream-select** protocol is used to negotiate which protocol to
-//! use on that given substream. In practice, Substrate opens the following substreams:
-//!
-//! - We periodically open an ephemeral substream in order to ping the remote and check whether the
-//! connection is still alive. Failure for the remote to reply leads to a disconnection. This uses
-//! the libp2p ping protocol.
-//! - We periodically open an ephemeral substream in order to ask information from the remote. This
-//! is called [the `identify` protocol](https://github.com/libp2p/specs/tree/master/identify).
-//! - We periodically open ephemeral substreams for Kademlia random walk queries. Each Kademlia
-//! query is done in a new separate substream. This uses the
-//! [standard libp2p Kademlia protocol](https://github.com/libp2p/specs/pull/108).
-//! - We optionally keep a substream alive for all Substrate-based communications. The name of the
-//! protocol negotiated is based on the *protocol ID* passed as part of the network configuration.
-//! This protocol ID should be unique for each chain and prevents nodes from different chains from
-//! connecting to each other. More information below.
-//!
-//! ## The Substrate substream
+//! a substream is open, the **multistream-select** protocol is used to negotiate which protocol
+//! to use on that given substream.
+//!
+//! Protocols that are specific to a certain chain have a `<protocol-id>` in their name. This
+//! "protocol ID" is defined in the chain specifications. For example, the protocol ID of Polkadot
+//! is "dot". In the protocol names below, `<protocol-id>` must be replaced with the corresponding
+//! protocol ID.
+//!
+//! > **Note**: It is possible for the same connection to be used for multiple chains. For example,
+//! >           one can use both the `/dot/sync/2` and `/sub/sync/2` protocols on the same
+//! >           connection, provided that the remote supports them.
+//!
+//! Substrate uses the following standard libp2p protocols:
```

Comment:
One could mention mdns here, no?

Reply:
I'm in the "Substreams" section, which talks about the substreams within TCP connections, so no. I mention mDNS in another section.